lepton/qemu.go

```diff
@@ -306,39 +306,91 @@ func parseQemuVersion(data []byte) string {
 	return rgx.FindString(string(data))
 }
 
-// kvmGroupPermission returns true if the current user is part of the kvm qroup,
-// false if she is not, or an error. or an error.
-func kvmGroupPermissions() error {
-	currentUser, err := user.Current()
+// kvmAvailable returns nil if the current user have read and write access to /dev/kvm
+// or error in other case
+func kvmAvailable() error {
+	u, err := user.Current()
 	if err != nil {
 		return err
 	}
-	if currentUser.Uid == "0" {
-		return nil
+
+	ok, err := userFilePermissions(u, "/dev/kvm", 6)
+	if err != nil {
+		return err
 	}
-	if inGroup, err := groupPermissions(currentUser, "kvm"); !inGroup || err != nil {
-		return &errQemuHWAccelNoUserRights{errCustom{"user not in kvm group", err}}
+	if !ok {
+		return &errQemuHWAccelNoUserRights{errCustom{"user have not read/write access to /dev/kvm", err}}
 	}
 	return nil
 }
 
-// groupPermissions returns true if the user is in the group specified by the
-// argument string, false if she is not, or an error.
-func groupPermissions(u *user.User, groupName string) (bool, error) {
-	gids, err := u.GroupIds()
+// userFilePermissions returns true if user have access to the file
+func userFilePermissions(u *user.User, name string, perm uint32) (bool, error) {
+	uid := u.Uid
+
+	// root have all access
+	if uid == "0" {
+		return true, nil
+	}
+
+	fi, err := os.Stat(name)
+	if err != nil {
+		return false, err
+	}
+
+	filePerm := fi.Mode().Perm()
+	ownerPerm := (filePerm & 0700) >> 6
+	groupPerm := (filePerm & 0070) >> 3
+	othersPerm := filePerm & 0007
+
+	// Now we need file stat to continue
+	fsys := fi.Sys()
+	fStatT, ok := fsys.(*syscall.Stat_t)
```

Comment:
this by in large looks great - the only comment I have is - isn't there a native go stat call you could use here instead of this?

Reply:
I've made changes according to @francescolavra comment